backend/ohq/views.py

```diff
@@ -774,3 +778,114 @@ def list(self, request, *args, **kwargs):
 
     def get_queryset(self):
         return Occurrence.objects.filter(pk=self.kwargs["pk"])
+
+class LlmSettingViewSet(viewsets.ModelViewSet, RealtimeMixin):
+    """
+    retrieve:
+    Return a prompt.
+
+    list:
+    Return a list of prompts specific to a queue.
+    Students can only see prompts they submitted.
+
+    create:
+    Create a prompt.
+
+    update:
+    Update all fields in the prompt.
+    You must specify all of the fields or use a patch request.
+
+    partial_update:
+    Update certain fields in the prompt.
+    Only specify the fields that you want to change.
+
+    destroy:
+    Delete a prompt.
+    """
+    permission_classes = [IsSuperuser]
+    serializer_class = LlmPromptSerializer
+
+    def get_queryset(self):
+        return LlmSetting.objects.filter(course=self.kwargs["course_pk"])
+    
+    def create(self, request, *args, **kwargs):
+        "Create a new LLM Prompt"
+        queryset = Course.objects.filter(pk=self.kwargs["course_pk"])
+        request.data["course"] = self.kwargs["course_pk"]
+        course = queryset[0]
+        request.data["llm_prompt"] = f"""
+            You are an AI TA designed to answer office hour questions for {course.department} {course.course_code}, which is a course on {course.course_title}. 
+            The description of the course is:
+            {course.description}
+            
+            Your responses should be informative and logical.
+
+            You should always refer to technical information.
+
+            Minimize any other prose.
+
+            Keep your answers short and impersonal.
+
+            YOU MAY ONLY ANSWER CONCEPTUAL QUESTIONS.
+
+            IF  YOU DO NOT FULLY UNDERSTAND THE USER'S QUESTION, ASK A FOLLOW UP QUESTION
+
+            YOU WOULD MUCH RATHER ANSWER TOO LITTLE THAN TOO MUCH.
+
+            Below are examples, where the question is deliminated in ticks, your response is in brackets, and each example is separated by commas:
+            'Q4?': [Please elaborate more on your question if you would like me to help]
+        """
+        return super().create(request, *args, **kwargs)
+    
+    def update(self, request, *args, **kwargs):
```

Comment:
You should implement this update method and it should not replace but only append to the current LLM prompt